### PR TITLE
V22F-453 recordings into tests

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Implement the recordings into tests so they can be ran and errors can be detected internally in Speed Dreams using an internal driver

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/rppUtils/RppUtils.hpp [hunk 1]
```diff
@@ -123,15 +123,16 @@ inline bool SetupSingletonsFolder()
...
 ///// Path must include file extension; no default extension is assumed.
 ///// Path is assumed to refer to an existing executable file
-inline void StartExecutable(const std::string& p_executablePath)
+inline void StartExecutable(const std::string& p_executablePath, const char* p_args = "")
 {
     // WARNING: This method of starting a process is Windows-exclusive.
     // Add a different method to run a process here if a Linux build is planned.
 
-    LPSTR args = _strdup("");                                       // Create an empty string of arguments for process
+    LPSTR args = _strdup(p_args);                                   // Create an empty string of arguments for process
```
Comment (noutheesink) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/rppUtils/RppUtils.hpp:135]: The comment on this line is now incorrect

**Review by noutheesink — CHANGES_REQUESTED** [review]

**Review by noutheesink — APPROVED** [review]
It looks like it will work. We should definitely all test it before merging it when everything is finally implemented

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/integrationTests/IntegrationTests.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+TEST_P(IntegrationTests, IntegrationTest)
+{
+    KillAllInterveningProcesses();
+    RunTest(GetParam());
+}
+
+/// @brief                  Instantiates the parameterized test
+/// INTEGRATION_TESTS_PATHS A CMake generated list of paths to different integration tests
+INSTANTIATE_TEST_CASE_P(
+    IntegrationTest,
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/integrationTests/IntegrationTests.cpp:165]:
```suggestion
    TestReplayRecordings,
```

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/integrationTests/IntegrationTests.cpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    KillProcessByName("speed-dreams-2.exe");
+}
+
+/// @brief The class that instantiates the different integration tests
+class IntegrationTests : public testing::TestWithParam<std::string>
+{
+};
+
+/// @brief The parameterized test, with all of the information needed for an integration test
+TEST_P(IntegrationTests, IntegrationTest)
```
Comment (larsdekwant) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/integrationTests/IntegrationTests.cpp:156]:
```suggestion
TEST_P(IntegrationTests, RunReplayRecording)
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]
Clarify test names.

**Review by larsdekwant — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/integrationTests/IntegrationTests.cpp [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    filesystem::path bbfile;
+
+    CheckFiles(p_path, bbfile);
+
+    std::string simulationArgs = GenerateSimulationArguments(p_path);
+
+    PROCESS_INFORMATION simulationInfo;
+    StartProcess(SD_EXECUTABLE, simulationArgs.c_str(), simulationInfo, SD_EXECUTABLE_WORKING_DIRECTORY);
+
+    //std::this_thread::sleep_for(std::chrono::seconds(5));
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/integrationTests/IntegrationTests.cpp:115]: Commented out code should be removed

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

**Review by suppergerrie2 — APPROVED** [review]

**Pull request merged**